awx/main/models/unified_jobs.py

```diff
@@ -810,11 +817,18 @@ def save(self, *args, **kwargs):
 
         # Sanity check: Has the job just completed? If so, mark down its
         # completion time, and record its output to the database.
+        #import sdb
+        #sdb.set_trace()
         if self.status in ('successful', 'failed', 'error', 'canceled') and not self.finished:
```

Comment:
The time that `status` changes to canceled is not the most important thing to me. I believe that will be at least partially duplicated information (particularly with finished).

What I think people should want is not the time that the job _actually_ cancels, but the time of the cancel request. For that, you want to capture the moment when the `cancel_flag` flips. That is very useful information because it tracks a user action, and can be used to diagnose situations where the system behavior... doesn't quite match what the user requested.